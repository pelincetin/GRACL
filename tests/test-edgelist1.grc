void printEdgeList(EdgeList el) {
    int i = el.length_EL();
    while (i > 0){
        Edge e = el.removeFirst_EL();
        print(e.start().data());
        el.appendEdge(e);
        i = i - 1;
    }
}

int main() {
    Graph g = createGraph();
        
    String hello = "Hello";
    String goodbye = "Goodbye";

    Node n1 = g.createNode(hello);
    Node n2 = g.createNode(goodbye);

    Edge e1 = g.addEdge(n1, n2, 15.3);
    Edge e2 = g.addEdge(n2, n1, 3.7);
    EdgeList el = createEdgeList();
    
    print("Empty should be true:");
    if (el.empty_EL()) {
        print("true");
    } else {
        print("false");
    }
    
    print("Append edge with start 'hello'");
    el.appendEdge(e1);
    print("Prepend edge with start 'goodbye'");
    el.prependEdge(e2);
    print("List should be goodbye hello:");
    printEdgeList(el);
}
