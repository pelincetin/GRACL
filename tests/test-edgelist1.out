Empty should be true:
true
Append edge with start 'hello'
Prepend edge with start 'goodbye'
List should be goodbye hello:
Goodbye
Hello